client/src/rpc_filter.rs

```diff
@@ -164,6 +185,82 @@ impl Memcmp {
     }
 }
 
+// Internal struct to hold Memcmp filter data as either encoded String or raw Bytes
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[serde(untagged)]
+enum DataType {
+    String(String),
+    Bytes(Vec<u8>),
```

Comment:
```suggestion
    Encoded(String),
    Raw(Vec<u8>),
```

nit: little more self-descriptive